iosApp/iosApp/app/AppDelegate.swift

```diff
@@ -98,10 +84,22 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 
     func applicationDidBecomeActive(_ application: UIApplication) {
         // Restart any tasks that were paused (or not yet started) while the application was inactive. If the application was previously in the background, optionally refresh the user interface.
+        serviceRegistry.notificationsApi.initializeNotifications(onSuccess: {(success:KotlinBoolean) -> KotlinUnit in
+            DispatchQueue.main.async {
```

Comment:
why the dispatch? Seems like it should already be in the main thread.

Reply:
When getNotificationSettings is called it may be executed on a background thread. We could move the dispatch to inside the initializeNotifications but it needs to be called somewhere.

Reply:
oh, interesting. I see that in the [docs](https://developer.apple.com/documentation/usernotifications/unusernotificationcenter/1649524-getnotificationsettings). I do think it makes sense dispatch back in the completion handler rather than out here if possible

Reply:
Moved in latest commit.